src/cfg.rs

```diff
@@ -1,314 +1,256 @@
 //  it under the terms of the GNU Affero General Public License as published by
 //  the Free Software Foundation, either version 3 of the License, or
 //  (at your option) any later version.
-//  
+//
 //  This program is distributed in the hope that it will be useful,
 //  but WITHOUT ANY WARRANTY; without even the implied warranty of
 //  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 //  GNU Affero General Public License for more details.
-//  
+//
 //  You should have received a copy of the GNU Affero General Public License
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
-use datatypes::Color;
+extern crate toml;
 
+use std::fs::File;
+use std::io::prelude::*;
+use std::{error,fmt,io,result};
+
+use datatypes::{Color,Palette};
+
+#[derive(Debug)]
+pub enum ConfigError {
+    Io(io::Error),
+    Parse(String), // TODO: once https://github.com/alexcrichton/toml-rs/issue#69
+                   // is closed, change this to Parse(toml::ParserError)
+}
+
+impl fmt::Display for ConfigError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ConfigError::Io(ref err) => write!(f, "IO Error: {}", err),
+            ConfigError::Parse(ref string) => write!(f, "{}", string),
+        }
+    }
+}
+
+impl error::Error for ConfigError {
+    fn description(&self) -> &str {
+        match *self {
+            ConfigError::Io(ref err) => err.description(),
+            ConfigError::Parse(ref string) => &string,
+        }
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
+        match *self {
+            ConfigError::Io(ref err) => err.cause(),
+            ConfigError::Parse(_) => None,
+        }
+    }
+}
+
+impl From<io::Error> for ConfigError {
+    fn from(err: io::Error) -> ConfigError{
+        ConfigError::Io(err)
+    }
+}
+
+pub type Result<T> = result::Result<T, ConfigError>;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
 pub struct Config {
-    pub font: &'static str,
+    pub font: String,
     pub scrollback: u32,
     pub tab_stop: u32,
     pub fg_color: Color,
     pub bg_color: Color,
     pub cursor_color: Color,
-    pub colors: [Color; 256]
+    pub colors: Palette,
+}
+
+impl Default for Config {
+    fn default() -> Config {
+        let source = include_str!("../resources/default-config.toml");
+        let table = parse_toml(&source.to_string(),
+                               &"../resources/default-config.toml".to_string()).unwrap();
+        Config::new_from_toml(&toml::Value::Table(table))
+    }
 }
 
-pub const CONFIG: Config = Config {
-    font: FONT,
-    scrollback: SCROLLBACK,
-    tab_stop: TAB_STOP,
-    fg_color: DEFAULT_FG,
-    bg_color: DEFAULT_BG,
-    cursor_color: CURSOR_COLOR,
-    colors: COLORS_256,
-};
+impl Config {
+    /// Constructs a new Config from a toml file specifed by the input string "path".
+    pub fn new_from_file(path: &String) -> Result<Config> {
```

Comment:
The signature of this function should be `new_from_file<P: AsRef<Path>>(path: P) -> Result<Config>`, and so should `read_toml_file` (see `std::fs::File::open` as an example). This signature means that you can pass a `String`, a `&str`, a `Path`, a `PathBuf`, whatever. It will also let you avoid a lot of type conversion methods elsewhere where you call this.